essentialkaos/go-confluence
# chore(api): support using personal tokes

See https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html

### What did you implement:

Support using personal access tokens

### How did you implement it:

Add new API function: `NewAPIWithToken`

### How can we verify it:

Create your API token and verify with the following example code:

```go
package main

import (
  "fmt"

  cf "github.com/essentialkaos/go-confluence/v5"
)

func main() {
  api, err := cf.NewAPIWithToken("https://confluence.domain.com", "your token")
  api.SetUserAgent("MyApp", "1.2.3")

  if err != nil {
    fmt.Printf("Error: %v\n", err)
    return
  }

  content, err := api.GetContentByID(
    "18173522", cf.ContentIDParameters{
      Version: 4,
      Expand:  []string{"space", "body.view", "version"},
    },
  )
  if err != nil {
    fmt.Printf("Error: %v\n", err)
    return
  }

  fmt.Printf("ID: %s\n", content.ID)
}
```

**Is this ready for review?:** Yes
**Is it a breaking change?:** No
